src/lib/ext-activitylog.js

```diff
@@ -46,6 +89,10 @@ class Controller {
 
   async init() {
     this.view.saveLogBtn.addEventListener('savelog', this);
+    this.view.extFilter.addEventListener('filterchange', this);
+    this.view.viewTypeFilter.addEventListener('filterchange', this);
+    this.view.apiTypeFilter.addEventListener('filterchange', this);
+    this.view.logView.modelFilter = (log) => this.isFilterMatched(log);
```

Comment:
`modelFilter` is too generic and not useful. These classes are representing concrete concepts, so try to use concrete names instead of generic abstract names.

The `logView` member of the `View` should be considered private. Don't assign to it, but add a new function that forwards the call:

```javascript
// In View
setLogFilter(filterFunc) {
  this.logView.setLogFilter(filterFunc);
}
```

And is it even meaningful to already assign the log filter here? The logic is only useful when the filter has been initialized. So perhaps you should defer the call until after the filters have been imported or updated.

The actual implementation very deep down in `log-view-element` could have a default filter `log => true` if you want to.

Reply:
done :-)
I have added a default filter function as false.